packages/ui/src/button/index.tsx

```diff
@@ -1,50 +1,52 @@
-import type { ButtonHTMLAttributes, PropsWithChildren } from "react";
-import IcVariant from "@/assets/icons/ic_variant.svg?react";
+import type { ButtonHTMLAttributes } from "react";
 import { cn } from "@/utils";
-import { buttonClasses } from "./button.variant";
+import { button, buttonText } from "./button.variant";
 
-export type ButtonProps = PropsWithChildren<
-  {
-    className?: string;
-    size: "md" | "lg";
-    icon?: boolean;
-    color?: "primary" | "secondary";
-  } & ButtonHTMLAttributes<HTMLButtonElement>
->;
+type CommonProps = {
+  className?: string;
+  icon?: React.ReactNode;
+  children: React.ReactNode;
+} & ButtonHTMLAttributes<HTMLButtonElement>;
+
+type ButtonBaseProps = {
+  size: "sm" | "md" | "lg" | "sticky";
+  variant: "primary" | "secondary" | "text";
+  selected?: never;
+  additionalText?: string;
+} & CommonProps;
+
+type ButtonRoundProps = {
+  size?: never;
+  variant: "round";
+  selected: boolean;
+  additionalText?: never;
+} & CommonProps;
+
+export type ButtonProps = ButtonBaseProps | ButtonRoundProps;
 
 export const Button = ({
   children,
   className,
   size,
-  icon = false,
-  color = "primary",
+  icon,
+  selected,
+  variant = "primary",
+  additionalText,
   ...props
 }: ButtonProps) => {
   return (
     <button
       className={cn(
-        "flex gap-2 rounded-[1.2rem] border-none transition-colors duration-200",
-        buttonClasses.size[size],
-        buttonClasses.variant[color].base,
-        buttonClasses.variant[color].hover,
-        buttonClasses.variant[color].focus,
-        buttonClasses.variant[color].active,
-        buttonClasses.variant[color].disabled,
+        button({ size, variant, ...(variant === "round" && { selected }) }),
```

Comment:
```suggestion
        button({ size, variant, selected }),
```

이 부분은 `compoundVariants`를 통해서 `round` 임을 코드에서 검증하지 않더라도, `selected`는 `variant`가 `round`일 때에만 유효한 스타일이 적용되어서 단순하게 줄일 수 있을 것 같아요.